src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -549,7 +547,34 @@ private void SearchPackageInActivePackageSource(string searchText, bool useCache
                     ? Resx.Resources.Text_Loading
                     : string.Format(CultureInfo.CurrentCulture, Resx.Resources.Text_Searching, searchText);
 
-                _packageList.LoadItems(loader, loadingMessage, _uiLogger);
+                // start SearchAsync task for initial loading of packages
+                var searchResultTask = loader.SearchAsync(null, CancellationToken.None);
+
+                _packageList.LoadItems(loader, loadingMessage, _uiLogger, searchResultTask);
+
+                // check if need to refresh update count
+                if (!useCache)
```

Comment:
whenever useCache is false, we'll always refresh update count. But there won't be a case where useCache is true but there isn't any cached results. And that's where we won't refresh update count.